auth.js

```diff
@@ -1,101 +1,60 @@
+        if (err) {
+          console.log(err, payload);
+          return next();
         }
-    })
-}catch(e){
-    next()
-}
-})
-
-  app.post('/api/world', (req, res) => {
-    console.log(req.body);
-    var testdata = new User({
-      username: req.body.username,
-      password: md5(req.body.password), 
-      email: req.body.email,
-      phone: req.body.phone,
-      question: req.body.question,
-      answer: req.body.answer
-    });
-    
-    testdata.save(function(err, data) {
-      if(err) console.log(err);
-      else res.json(data);
-    });
+        User.findById(payload.userId).then(doc => {
+          req.user = doc;
+          next();
+        });
+      });
+    } catch (e) {
+      next(e);
+    }
   });
 
   // -------- Get for user login ----------
 
-  app.post('/api/login', async (req, res) => {
-    //console.log('request header: ', req.header);
-    //console.log('PRINTED req body', req.body);
-    var reqEmail = req.body.email;                               //  set the username the email the user types in
-    let currentUser = await User.find({ email: reqEmail });      //  only bring back that email
-    if (currentUser.length == 0) {                               // check the database for the email entered
-      //console.log("That email does not exist. If you don't have an account, sign up!");
-      res.json({message: 'none'});
-    }else{
-     
-      var dbPassword = currentUser[0].password;
-      var newPW = md5(req.body.password)              //hash what the user typed in
-
-      if (checkPW(newPW, dbPassword)){
-        //var date = new Date();
-        var ten_minutes = moment().add(10, 'm');
-        const token=jwt.sign({expires: ten_minutes, userId:currentUser[0]._id}, "thisisaTOKENKEY");
-          User.findById(currentUser[0]._id).populate("modals")  
-            .then(data => {
-              res.json({
-                userId: currentUser[0]._id,
-                // username:user.username,
-                // image:user.image,
-                // name:user.first,
-                expires: ten_minutes,
-                token,
-                data                  // this data should contain the notes (modals) attached to this user.
-            })
-            })
-                    
-      }else{
-        res.status(400).json({message:'Invalid Password/Username'});
-      }                     
-      
-    }                                        
-  });
-  
-}
-
-
+  app.post("/api/login", async (req, res) => {
+    const currentUser = (await User.find({ email: req.body.email }))[0] || null;
+    if (!currentUser) {
+      return res.status(400).json({ message: "none" });
+    }
 
+    const hashedPassword = md5(req.body.password);
 
+    if (!checkPW(hashedPassword, currentUser.password)) {
+      return res.status(400).json({ message: "Invalid Password/Username" });
+    }
+    const token = jwt.sign({ userId: currentUser._id }, SECRET, {
+      expiresIn: 600
```

Comment:
This might be one of the major changes. I hope this isn't 600ms though lol. Check this and make sure it's the right length, but it keep it in seconds or whatever if you can instead of ten_minutes, then you can try that once you know it works otherwise.